src/widgets/sparkline.rs

```diff
@@ -158,52 +336,85 @@ impl Sparkline<'_> {
         if spark_area.is_empty() {
             return;
         }
-
-        let max = self
+        // determine the maximum height across all bars
+        let max_height = self
             .max
-            .unwrap_or_else(|| *self.data.iter().max().unwrap_or(&1));
+            .unwrap_or_else(|| self.data.iter().filter_map(|s| s.bar).max().unwrap_or(1));
+
+        // determine the maximum index to render
         let max_index = min(spark_area.width as usize, self.data.len());
-        let mut data = self
-            .data
-            .iter()
-            .take(max_index)
-            .map(|e| {
-                if max == 0 {
-                    0
-                } else {
-                    e * u64::from(spark_area.height) * 8 / max
+
+        // render each item in the data
+        for (i, item) in self.data.iter().take(max_index).enumerate() {
+            let x = match self.direction {
+                RenderDirection::LeftToRight => spark_area.left() + i as u16,
+                RenderDirection::RightToLeft => spark_area.right() - i as u16 - 1,
+            };
```

Comment:
This for loop can probably be simplified to something like:
```rust
let cols = if self.direction == RenderDirectionLeftToRight { area.cols() } else { area.cols.rev() };
for (item, col) in zip(self.data, cols) {
   ...
}
```

Reply:
I tried with:

```rust
let cols = if self.direction == RenderDirection::LeftToRight {
    spark_area.columns()
} else {
    spark_area.columns().rev()
};
for (item, col) in self.data.iter().take(max_index).zip(cols) { 
  ...snip...
  buf[(col.x, spark_area.top() + j)]
      .set_symbol(symbol)
      .set_style(self.style.patch(style.unwrap_or_default()));
}
```

However it fails with `rev()` as:

```
the trait `DoubleEndedIterator` is not implemented for `rect::iter::Columns`
```

Reply:
Makes sense. That's addable, but not necessary for this PR.